src/player.py

```diff
@@ -53,9 +58,23 @@ def update(self, delta_time):
                 * delta_time
             )
 
-        self.primary_weapon.update(delta_time)
-        self.secondary_weapon.update(delta_time)
+        self.primary_weapon.update(delta_time,self.projectile_list, self.beam_list)
+        self.secondary_weapon.update(delta_time,self.projectile_list, self.beam_list)
+        for projectile in self.projectile_list:
+            projectile.center_x += projectile.change_x * delta_time
+            projectile.center_y += projectile.change_y * delta_time
```

Comment:
Fine for now, a note for the future:
If projectiles want to implement their own movement logic, for example, a rocket that accelerates or does "Itano circus" nonsense, then we can move this logic onto the projectile.  That way different projectiles can have different behaviors, and `Player` can remain agnostic.

Reply:
Agreed, I'll do this in the near future